04_flora_cnell/src/plot_utils.R

```diff
@@ -0,0 +1,214 @@
+
+plot_spring_index <- function(spring_timing, proj, file_out){
+  
+
+  # access Contiguous U.S. state map:
+  state_map <- spData::us_states %>% st_transform(proj)
+  
+  spring_timing %>%
+    ggplot() +
+    geom_sf(data = state_map, fill = NA, color = "grey", alpha = 0.5) +
+    geom_tile(aes(x =x, y = y, fill = timing_day),
+              alpha = 0.8)+
+    geom_sf(data = state_map%>%st_union(), fill = NA, color = "grey") +
+    theme_void()+
+    theme(
+      plot.background = element_rect(fill = NA, color = NA),
+      panel.background = element_rect(fill = NA, color = NA),
+      plot.title = element_text(color = "white", size = 30, face = "bold"),
+      plot.subtitle = element_text(color = "white"),
+      legend.position = 'none')+
+    coord_sf()+
+    scale_fill_stepsn(colors = sequential_hcl(10, "TealGrn"),
+                      "First leaf date",
+                      na.value = NA,
+                      show.limits = TRUE,
+                      n.breaks =10)+
+    ggtitle("Spring has sprung in many parts of the conterminous U.S.",
+            subtitle = "Data: USA National Phenology Network. 2022. First Leaf - Spring Index. as of 04/03/2022 for conterminous U.S. USA-NPN, Tucson, Arizona, USA. http://dx.doi.org/10.5066/F7SN0723")+
+    guides(fill = guide_colorsteps(
+      direction = "horizontal",
+      title.position = "top",
+      label.theme = element_text(color = "white"),
+      title.theme = element_text(color = "white"),
+      barwidth = 15
+    ))
+  
+  ggsave(file_out, width = 16, height = 9)
+  return(file_out)
+}
+plot_spring_steps <- function(spring_timing, hist_timing, proj, file_out){
+  # create stepped histogram
+  # number of cells in each day
+  anom_hist <- spring_timing %>% 
```

Comment:
I think you could pull the first three lines of this munging out into a helper function in `data_utils.R` that you could then call for both `spring_timing` and `hist_timing`. Then the only bit left in the plotting function would be the 3 lines to add rows to `spring_timing`, which is for plotting purposes.